clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -289,4 +291,18 @@ public boolean hasExpired() {
     synchronized List<KafkaMetric> metrics() {
         return Collections.unmodifiableList(new LinkedList<>(this.metrics.values()));
     }
+
+    private Object metricLock(Stat stat) {
```

Comment:
We also instantiate KafkaMetric in Metrics.addMetric(). Perhaps we can add a comment there that the user is expected to add synchronization in the exposed objects, if needed.